marlowe/test/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -419,6 +419,94 @@ choiceNotInBounds bounds =
     suchThat arbitrary $ \chosenNum -> not $ any (inBound chosenNum) bounds
 
 
+-- | Generate relevant Input content for a given input action
+interestingInput :: Bool -> Action -> [InputContent]
+interestingInput validity (Choice choiceId) bounds =
+  map (IChoice choiceId) (interestingChoiceNum bounds)
+  ++ map (\choiceId -> IChoice choiceId bounds) (interestingChoiceId choiceId)
+interestingInput validity (Choice choiceId) bounds = IChoice choiceId <$> interestingChoiceNum' validity bounds
+
+interestingChoiceId :: ChoiceId -> [ChoiceId]
+interestingChoiceId _ = error "not implemented yet"
+
+interestingParties :: Party -> [Party]
+interestingParties (Role "Alice") = [Role "Alice", Role "Alic", Role "Alices"]
+
+interestingChoiceNum :: [Bound] -> [ChoiceNum]
+interestingChoiceNum bounds = concatMap interestingChoiceNum' bounds
+
+interestingChoiceNum' :: Bool -> Bound -> [ChoiceNum]
+interestingChoiceNum' True (Bound lower upper)
+  | lower > upper = []
+  | lower == upper = [lower]
+  | otherwise = Data.List.nub validValues lower upper
+
+interestingChoiceNum' False (Bound lower upper)
+  | lower > upper = Data.List.nub allValues lower upper
+  | otherwise = Data.List.nub invalidValues lower upper
+
+validValues :: Int -> Int -> [Int]
```

Comment:
We might want to give this a slightly different name, since if `lower == upper`, it *doesn't* produce valid values. Or there could be guards in the functions it calls.